### ISSUE TITLE
[Feature Request] Button accessibility issue

### ISSUE THREAD

**Issue by Elfayer** [vuetifyjs/vuetify#11937]
### Problem to solve
When I try: `<v-btn>Submit</v-btn>`
I get a Lighthouse Accessibility error: `Buttons do not have an accessible name`
Because the output is:
```html
<button type="button" class="v-btn v-btn--contained theme--light v-size--default">
  <span class="v-btn__content">Submit</span>
</button>
```

### Proposed solution
Why not add the text of the button also as `aria-label` on the `<button>`?
It seems like it'd solve the problem: https://web.dev/button-name/?utm_source=lighthouse&utm_medium=devtools#how-to-add-accessible-names-to-buttons

**Comment by KaelWD** [comment]
> For buttons with visible labels, add text content to the button element. 

It should have either text content or aria-label, not both.

**Comment by Elfayer** [comment]
Thing is, since Vuetify adds a span in the button, to a reader it has no text. (At least if we trust the Lighthouse)

**Comment by KaelWD** [comment]
Might want to report that to https://github.com/GoogleChrome/lighthouse then, I've never heard of it being a problem.

**Comment by kaovilai** [comment]
@KaelWD 
Consider re-opening this issue because the accessibility here is regarding screen readers ability to follow default HTML semantics.

I recently received the following _How to fix_ from [microsoft/accessibility-insights-web](https://github.com/microsoft/accessibility-insights-web)
```
Fix ONE of the following:
Element does not have inner text that is visible to screen readers
aria-label attribute does not exist or is empty
aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
Element's default semantics were not overridden with role="none" or role="presentation"
Element has no title attribute
```

This is not a problem with lighthouse either since the expectation from the [HTML standard regarding semantics is pretty clear.](https://www.w3schools.com/html/html_accessibility.asp)

It is not high priority for most of us but this issue should not be left closed.

**Comment by KaelWD** [comment]
This is remaining closed unless you can demonstrate that it actually causes problems with screen readers.

**Comment by kaovilai** [comment]
It's simply good practice like [lighthouse team have said in an article](https://web.dev/button-name/). Do you think the lighthouse team is wrong?

Most accessibility screen readers can probably cope with a single span or div since it's a widespread issue but would probably struggle to select the right text to do Text-To-Speech on if there are more than one span or div.

I am not referring to article readers but rather Disability.

For more empathy stories see [w3.org](https://www.w3.org/WAI/people-use-web/user-stories/#accountant)

> ### Ilya, senior staff member who is blind
> Ilya is blind. She is the chief accountant at an insurance company that uses web-based documents and forms over a corporate intranet and like many other blind computer users, she does not read Braille.
> 
> Ilya uses a screen reader and mobile phone to access the web. Both her screen reader and her mobile phone accessibility features provide her with information regarding the device’s operating system, applications, and text content in a speech output form.

**Comment by kaovilai** [comment]
Sounds like a good-first-issue.

**Comment by KaelWD** [comment]
> Do you think the lighthouse team is wrong?

No, but their heuristics for detecting this might be. "text content" to me means innerText, which is set when there's an extra span around it:

```
<button><span>text content</span></button>

el.innerText // "text content"
```

If you can demonstrate a screen reader NOT reading this correctly then the issue can be reopened. 

> Why not add the text of the button also as aria-label on the `<button>`?

Can't really do that automatically. You're free to add aria-label yourself if you want to follow lighthouse without verification.

**Comment by kaovilai** [comment]
Since we know there is going to be a <span> inside, we can programmatically assign the <button> `aria-labelledby` attribute. But I can see how it could get problematic with the need for ids for each button's span.

`aria-label` is easy enough if you care about it enough. But it would just be nice for everyone else to also benefit in the background.

Thanks for input on the issue.

**Comment by kaovilai** [comment]
If the screen reader is following the [ARIA](https://www.w3.org/TR/wai-aria/) spec exactly and may not do its own propagation to inner elements.
